// math.gl/dist-es6/gl-matrix-min/vec3.js
// Copyright (c) 2017 Uber Technologies, Inc.
//
// Permission is hereby granted, free of charge, to any person obtaining a copy
// of this software and associated documentation files (the "Software"), to deal
// in the Software without restriction, including without limitation the rights
// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
// copies of the Software, and to permit persons to whom the Software is
// furnished to do so, subject to the following conditions:
//
// The above copyright notice and this permission notice shall be included in
// all copies or substantial portions of the Software.
//
// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
// THE SOFTWARE.

// gl-matrix is too big. Cherry-pick individual imports from stack.gl version
/* eslint-disable camelcase */
export { default as vec3_set } from 'gl-vec3/set';
export { default as vec3_length } from 'gl-vec3/length';
export { default as vec3_distance } from 'gl-vec3/distance';
export { default as vec3_angle } from 'gl-vec3/angle';
export { default as vec3_add } from 'gl-vec3/add';
export { default as vec3_subtract } from 'gl-vec3/subtract';
export { default as vec3_multiply } from 'gl-vec3/multiply';
export { default as vec3_divide } from 'gl-vec3/divide';
export { default as vec3_scale } from 'gl-vec3/scale';
export { default as vec3_scaleAndAdd } from 'gl-vec3/scaleAndAdd';
export { default as vec3_negate } from 'gl-vec3/negate';
export { default as vec3_inverse } from 'gl-vec3/inverse';
export { default as vec3_normalize } from 'gl-vec3/normalize';
export { default as vec3_dot } from 'gl-vec3/dot';
export { default as vec3_cross } from 'gl-vec3/cross';
export { default as vec3_lerp } from 'gl-vec3/lerp';
export { default as vec3_rotateX } from 'gl-vec3/rotateX';
export { default as vec3_rotateY } from 'gl-vec3/rotateY';
export { default as vec3_rotateZ } from 'gl-vec3/rotateZ';
